### ISSUE TITLE
dict(FlatDict) fails when it contains complex unicode keys

### ISSUE THREAD

**Issue by iliapolo** [gmr/flatdict#19]
There seem to be various issues when I put a complex unicode key inside the flatdict. 

For example:

```python
import flatcdict
d = flatdict.Flatdict()
d[u'key1:key2'] = 'value1'
dict(d)

KeyError: u'key1'
```

The same thing works if the key is a string object
```python
import flatcdict
d = flatdict.Flatdict()
d['key1:key2'] = 'value1'
dict(d)

{'key1:key2': 'value1'}
```
I tracked the issue to [this line](https://github.com/gmr/flatdict/blob/master/flatdict.py#L364) where you force the key to be an 'str' object for it to have a delimiter. 

Any special reason for that?

```python
def _has_delimiter(self, key):
    """Checks to see if the key contains the delimiter.

    :rtype: bool

    """
    return isinstance(key, str) and self._delimiter in key
```

Its a problem for me because I populate the flatdict with objects returned by the ```json.loads``` function, which uses **unicode** for keys.

**Environment**

Python 2.7.14 on macOS HighSierra 10.13.3

I'll be happy to contribute and fix this issue once we discuss it. 

Thanks 👍

**Comment by iliapolo** [comment]
Well, I just realised that is Python 3, **unicode** was renamed to **str**. So is it just a compatibility issue with Python 2.7?

**Comment by gmr** [comment]
@iliapolo probably, I'll take a look.

**Comment by gmr** [comment]
This should be resolved in v3.0.1
